sdks/java/io/cdc/src/main/java/org/apache/beam/io/cdc/DebeziumOffsetTracker.java

```diff
@@ -25,28 +25,46 @@
 
 import java.util.Map;
 
+/**
+ * {@link RestrictionTracker} for Debezium connectors
+ */
 public class DebeziumOffsetTracker extends RestrictionTracker<DebeziumOffsetHolder, Map<String, Object>> {
     private static final Logger LOG = LoggerFactory.getLogger(DebeziumOffsetTracker.class);
 
     private DebeziumOffsetHolder restriction;
     private static final long MILLIS = 60 * 1000;
+    private final Integer maxRecords = 10;
 
     DebeziumOffsetTracker(DebeziumOffsetHolder holder) {
         this.restriction = holder;
     }
 
+    /**
+     * Overriding {@link #tryClaim} in order to stop fetching records from the database.
+     *
+     * <p>Currently this works time-based:</p>
+     * </p>
+     * User may specify the amount of time the connector to be kept alive.
+     * Please see {@link KafkaSourceConsumerFn} for more details on this.
+     * </p>
+     *
+     * @param position Currently not used
+     * @return boolean
+     */
     @Override
     public boolean tryClaim(Map<String, Object> position) {
         LOG.debug("-------------- Claiming {} used to have: {}", position, restriction.offset);
         long elapsedTime = System.currentTimeMillis() - KafkaSourceConsumerFn.startTime.getMillis();
+        int fetchedRecords = this.restriction.fetchedRecords == null ? 0 : this.restriction.fetchedRecords + 1;
         LOG.debug("-------------- Time running: {} / {}", elapsedTime, (KafkaSourceConsumerFn.minutesToRun * MILLIS));
-        this.restriction = new DebeziumOffsetHolder(position, this.restriction.history);
+        this.restriction = new DebeziumOffsetHolder(position, this.restriction.history, fetchedRecords);
+        LOG.debug("-------------- History: {}", this.restriction.history);
         if (KafkaSourceConsumerFn.minutesToRun < 0) {
-            return true;
+            return fetchedRecords < maxRecords;
         }
-        return elapsedTime < (KafkaSourceConsumerFn.minutesToRun * MILLIS);
+        return (elapsedTime < (KafkaSourceConsumerFn.minutesToRun * MILLIS)) || (fetchedRecords < maxRecords);
```

Comment:
Can you add a unit test module for this `DebeziumOffsetTracker ` class and verify (test) those scenarios (time-based, count-based, infinite)

Reply:
Done